#164892850 social auth facebook and google

ssanusi opened the pull request:
#### Description
The Pull Request add the feature of authentication on the app via facebook and google 

#### Type of change
 - added Passport-facebook and Passport-google for social authentication

#### How Has This Been Tested?
 - navigate to the route ```/api/v1/auth/facebook``` you will be required to log in to the application 
 - navigate to the route ```/api/v1/auth/google``` you will be required to log in to the application 

#### Checklist:
 - [x] Facebook Authentication
 - [x] Google Authentication
 - [x] Unit Testing
 - [x] Integration Testing

#### PT
[#164892850](https://www.pivotaltracker.com/story/show/164892850)

ssanusi committed:
164892850-feature(auth) Implement facebook authentication
- implement facebook authentication

[delivers  #164892850]

ssanusi committed:
164892850-feature(auth) Implement facebook authentication
- implement google authentication

[delivers  #164892850]

chikeozulumba commented on src/routers/index.js:
```diff
@@ -1,9 +1,30 @@
 router.get('/', (req, res) => res.status(200).json({
   message: 'Welcome to the Authors Haven API',
-}));
+}),);
```
@ssanusi Nice job done, you can review the flags raised by Hound CI.

chikeozulumba commented on tests/user.test.js:
```diff
@@ -0,0 +1,30 @@
+import chai from 'chai';
```
@ssanusi Integration tests and unit tests are yet to be done

ssanusi replied:
@chikeozulumba its work in progress thanks for pointing that out i have to figure out how the unit test will be implemented

ssanusi committed:
164892850-feature(auth) Implement facebook authenticatio
- refactor code

[delivers  #164892850]

ssanusi committed:
164892850-feature(auth) Implement facebook authentication
- refactor

[delivers  #164892850]

164892850-feature(auth) refactor

-refactor

[delivers  #164892850]

ssanusi committed:
164892850-feature(auth) refactor

-refactor

[delivers  #164892850]

ssanusi committed:
164892850-feature(auth) refactor

-refactor

[delivers  #164892850]

ssanusi committed:
164892850-feature(auth) refactor

-refactor

[delivers  #164892850]
